### PR TITLE
master-changes

### PR DESCRIPTION
**Pull request by harjaapdhillon16**
Made some changes in the contract address files to get the code ready for prod

### CODE REVIEW

File: src/utils/contract-addresses.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
-export const app_contract = 'registry-1.i-am-human.testnet';
+export const app_contract =
+  process.env.REACT_APP_ENV === 'dev'
+    ? 'registry-1.i-am-human.testnet'
+    : 'registry.i-am-human.near';
 export const new_sbt_contract = 'sbt1.i-am-human.testnet';
-
+console.log(process.env.REACT_APP_ENV);
```
Comment (Megha-Dev-19) [inline src/utils/contract-addresses.js]: we can remove it, right

File: src/utils/near-wallet.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ export class Wallet {
     // Omitting the accountId will result in the user being
     // asked to sign all transactions.
     this.createAccessKeyFor = createAccessKeyFor;
-    this.network = 'testnet';
+    this.network = process.env.REACT_APP_ENV === 'dev' ? 'testnet' : 'mainnet';
```
Comment (Megha-Dev-19) [inline src/utils/near-wallet.js]: why do we need testnet now, if we are moving to prod, then irrespective of local or prod, mainnet should be used
If we (devs) need to check something, we can manually do it
we should add a config file for such things (to carry all types of contract)
for eg: https://github.com/alpha-fi/cheddar-tic-tac-toe-ui/blob/master/src/near/config.ts

File: src/utils/supabase.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import axios from 'axios';
 
 export const api_link =
-  process.env.NODE_ENV === 'development'
-    ? 'http://localhost:3001'
-    : 'https://dev-ophc7vkxsq-uc.a.run.app';
+  process.env.REACT_APP_ENV === 'dev'
+    ? 'https://dev-ophc7vkxsq-uc.a.run.app'
+    : 'https://api-ophc7vkxsq-uc.a.run.app';
```
Comment (Megha-Dev-19) [inline src/utils/supabase.js]: same thing

**Review by Megha-Dev-19 — CHANGES_REQUESTED** [review]
Let's create a separate file with all the contracts config and use that

File: src/utils/contract-addresses.js (hunk 1 above)
Comment (harjaapdhillon16) [reply src/utils/contract-addresses.js]: yep removing

File: src/utils/near-wallet.js (hunk 2 above)
Comment (harjaapdhillon16) [reply src/utils/near-wallet.js]: This isn't a contract it's just a config switching mainnet and testnet , and manually changing the config isn't a good practice with the current approach you can just switch the env variable and every credential changes from dev to prod

File: src/utils/supabase.js (hunk 3 above)
Comment (harjaapdhillon16) [reply src/utils/supabase.js]: Same comment as above

File: src/utils/near-wallet.js (hunk 2 above)
Comment (Megha-Dev-19) [reply src/utils/near-wallet.js]: but there are the changes related to the network config right?

File: src/utils/supabase.js (hunk 3 above)
Comment (Megha-Dev-19) [reply src/utils/supabase.js]: this is related to the backend API right which should also be part of config

File: src/components/pages/landing/Gooddollar.js [hunk 4]
```diff
@@ -140,6 +140,7 @@ export const Gooddollar = ({ setShowGooddollarVerification }) => {
       try {
         if (data.error) return alert('Login request denied !');
         parseLoginResponse(data).then((d) => {
+          console.log(d);
```
Comment (Megha-Dev-19) [inline src/components/pages/landing/Gooddollar.js]: can we remove it?

**Review by Megha-Dev-19 — COMMENTED** [review]
I don't see a config file yet
The current issue is if I m new developer I will have to go through the struggle of finding where can i move from mainnet to testnet

- So we should add the config file (that is the basic practice followed in majority)

- and also add a readme about it, so when the user installs this repo on their local they know how to make the changes 

- do not do more unfamiliar changes in the existing PR, create a new PR 
and add some valid title and description for it

File: src/utils/near-wallet.js (hunk 2 above)
Comment (harjaapdhillon16) [reply src/utils/near-wallet.js]: resolved it to the contract-addresses file

File: src/utils/contract-addresses.js [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
-  process.env.REACT_APP_ENV === 'dev'
-    ? 'og-sbt-1.i-am-human.testnet'
-    : 'og-sbt.i-am-human.near';
-export const gooddollar_contract =
-  process.env.REACT_APP_ENV === 'dev'
-    ? 'gooddollar-v1.i-am-human.testnet'
-    : 'gooddollar-v1.i-am-human.near';
+import { getEnv } from './config';
+
+const environment = process.env.REACT_APP_ENV;
```
Comment (Megha-Dev-19) [inline src/utils/contract-addresses.js]:
```suggestion
const environment = process.env.REACT_APP_ENV ?? "prod";
```

**Review by Megha-Dev-19 — APPROVED** [review]

**Pull request merged**